apps/api/src/domain/models/media-item/commands/publish-media-item.command.integration.spec.ts

```diff
@@ -0,0 +1,193 @@
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA: () => validCommandFSA,
+                initialState,
+                checkStateOnSuccess: async () => {
+                    const result = await testRepositoryProvider
+                        .forResource(ResourceType.mediaItem)
+                        .fetchById(validCommandFSA.payload.id);
+
+                    expect(result).toBeInstanceOf(MediaItem);
+
+                    const updatedInstance = result as MediaItem;
+
+                    const { id, published } = updatedInstance;
+
+                    expect(id).toBe(unpublishedMediaItem.id);
+
+                    expect(published).toBe(true);
+
+                    expect(
+                        updatedInstance.eventHistory.some(
+                            ({ type }) => type === 'MEDIA_ITEM_CREATED'
+                        )
+                    );
+                },
+            });
+        });
+    });
+
+    describe('when the command payload has an invalid type', () => {
+        describe('when the id is missing', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => ({
+                        type: commandType,
+                        payload: {},
+                    }),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
+                    },
+                });
+            });
+        });
+
+        describe('when the id is a number', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => buildInvalidCommandFSA(767),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
+                    },
+                });
+            });
+        });
+
+        describe('when the id is an array', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => buildInvalidCommandFSA(['12', '123']),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
```

Comment:
Interesting, I'd like to see `innerErrors` here just to be sure there's no possibility that another word has `id` in it.

Reply:
You have to understand that these messages are generated by the validation library. They are not specific Coscrad custom errors. I figure this is more specific than simply checking `instanceof InvalidCommandPayloadTypeError` on the outer error.